### PR TITLE
Cookbook

### PR DESCRIPTION
**Pull request by priverop**
### Description
As described here https://github.com/SceneGate/Yarhl/issues/105 we are creating a new article for the docs called "Cookbook", which will have examples of use. We will also add the most asked questions.

### WIP

- [x] Add article "Cookbook"
- [x] Parameters in converters
- [x] Padding examples
- [x] Virtual directory structure
- [x] Iterating children nodes
- [x] Searching for nodes
- [x] Deleting nodes

### CODE REVIEW

**Comment by priverop** [comment]
Parameter in a Converter:  https://github.com/pleonex/PokemonConquest/blob/master/AmbitionConquest/Texts/Binary2TextList.cs

**Comment by priverop** [comment]
Finished @pleonex. Please review the examples, thanks!

File: docs/articles/Cookbook.md [hunk 1]
```diff
@@ -3,33 +3,90 @@
 ## IO Examples
 ### Padding
 ```csharp
+    // Add 0s until position is divisible by 4
+    filesWriter.WritePadding(0x00, 0x4);
+    // Add 0s until position is divisible by 10
```
Comment (pleonex) [inline docs/articles/Cookbook.md]: Let's put the `10` in hex: by `0x10` or by 16.

Comment (priverop) [reply docs/articles/Cookbook.md]: Fixed, thanks!

**Pull request merged**